src/editors/EditorHeader.jsx

```diff
@@ -1,44 +1,112 @@
 import {
-  ActionRow, IconButton, Icon, ModalDialog,
+  ActionRow, IconButton, Icon, ModalDialog, Form,
 } from '@edx/paragon';
-import PropTypes from 'prop-types';
-import React, { useContext } from 'react';
-import { Close } from '@edx/paragon/icons';
+import React, { useContext, useState, useEffect } from 'react';
+import { Edit, Close } from '@edx/paragon/icons';
 import EditorPageContext from './EditorPageContext';
-import { ActionStates, mapBlockTypeToName } from './data/constants';
+import { ActionStates } from './data/constants';
 
-const EditorHeader = ({ title }) => {
-  const { unitUrl, unitUrlLoading, studioEndpointUrl } = useContext(EditorPageContext);
+/* The Header component renders based on three states:
+    loading -> loaded but not editing -> editing. */
+
+const EditorHeader = () => {
+  const {
+    titleRef, editorRef, setBlockTitle,
+    blockValue, blockLoading,
+    unitUrl, unitUrlLoading, studioEndpointUrl,
+  } = useContext(EditorPageContext);
+
+  const [editing, setEditing] = useState(false);
+
+  const [title, setTitle] = useState('');
+  useEffect(() => {
+    if (blockLoading === ActionStates.FINISHED) {
+      setTitle(blockValue ? blockValue.data.display_name : '');
+    }
+  }, [blockLoading]);
+
+  const handleChange = (e) => {
+    setTitle(e.target.value);
+  };
+
+  const handleKeyDown = (e) => {
+    if (e.key === 'Enter') { setEditing(false); }
+    if (e.key === 'Tab' && editorRef) {
+      e.preventDefault();
+      editorRef.current.focus();
+    }
+  };
```

Comment:
Handles various key inputs from user. Pressing 'tab' while editing the title will bring the cursor to the editor.

Reply:
What behavior does onBlur accomplish, I'm not sure why Blur performs the tasks required when "I'm done editing"

Reply:
I intended for onChange and onKeyDown to update local state while onBlur updates the state in the provider. I think putting it all in onKeyDown makes sense. I'll update the code for this.

Reply:
On re-examining this, onBlur covers the case of when the user clicks out of the edit.

Reply:
Ok, that makes sense I guess if it accomplishes the task.